python/examples/extract_position_data.py

```diff
@@ -14,30 +14,82 @@
 from fusion_engine_client.utils.argument_parser import ArgumentParser
 from fusion_engine_client.utils.log import locate_log, DEFAULT_LOG_BASE_DIR
 
-KML_TEMPLATE = """\
+KML_TEMPLATE_START = """\
 <kml xmlns="http://www.opengis.net/kml/2.2">
   <Document>
-    <Placemark>
-      <name>Position</name>
-      <Style>
-        <LineStyle>
-          <width>8</width>
-          <color>#ff0000ff</color>
-        </LineStyle>
-      </Style>
-      <LineString>
-        <extrude>0</extrude>
-        <tesselate>1</tesselate>
-        <altitudeMode>clampToGround</altitudeMode>
-        <coordinates>
-%(coordinates)s
-        </coordinates>
-      </LineString>
-    </Placemark>
+    <name>FusionEngine Trajectory</name>
+    <Style id="type-0">
+      <IconStyle>
+        <scale>0.3</scale>
+        <color>FF000000</color>
+        <Icon>
+          <href>https://maps.google.com/mapfiles/kml/shapes/road_shield3.png</href>
+        </Icon>
+      </IconStyle>
+    </Style>
+    <Style id="type-1">
+      <IconStyle>
+        <scale>0.3</scale>
+        <color>FF0000FF</color>
+        <Icon>
+          <href>https://maps.google.com/mapfiles/kml/shapes/road_shield3.png</href>
+        </Icon>
+      </IconStyle>
+    </Style>
+    <Style id="type-2">
+      <IconStyle>
+        <scale>0.3</scale>
+        <color>FFFF0000</color>
+        <Icon>
+          <href>https://maps.google.com/mapfiles/kml/shapes/road_shield3.png</href>
+        </Icon>
+      </IconStyle>
+    </Style>
+    <Style id="type-4">
+      <IconStyle>
+        <scale>0.3</scale>
+        <color>FF00A5FF</color>
+        <Icon>
+          <href>https://maps.google.com/mapfiles/kml/shapes/road_shield3.png</href>
+        </Icon>
+      </IconStyle>
+    </Style>
+    <Style id="type-5">
+      <IconStyle>
+        <scale>0.3</scale>
+        <color>FF008000</color>
+        <Icon>
+          <href>https://maps.google.com/mapfiles/kml/shapes/road_shield3.png</href>
+        </Icon>
+      </IconStyle>
+    </Style>
+    <Style id="type-6">
+      <IconStyle>
+        <scale>0.3</scale>
+        <color>FFFFFF00</color>
+        <Icon>
+          <href>https://maps.google.com/mapfiles/kml/shapes/road_shield3.png</href>
+        </Icon>
+      </IconStyle>
+    </Style>
+"""
+
+KML_TEMPLATE_END = """\
   </Document>
 </kml>
 """
 
+KML_TEMPLATE = """\
+    <Placemark>
+      <Timestamp>%(timestamp)s</Timestamp>
+      <styleUrl>#type-%(solution_type)s</styleUrl>
```

Comment:
use d instead of s, then you don't have to convert to `str()` below, you can just use the integer directly.